modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -853,16 +875,39 @@ class StreamingOutput final: public cv::gimpl::GIslandExecutable::IOutput
     }
 };
 
+struct IslandExecutableData {
+    std::shared_ptr<cv::gimpl::GIslandExecutable> island_exec;
+    std::string island_meta_info;
+};
+
+std::string buildIslandName(const ade::NodeHandle& op_nh,
+                            const cv::gimpl::GIslandModel::Graph& gim) {
+    std::string island_name = gim.metadata(op_nh).get<cv::gimpl::FusedIsland>().object->name();
+
+    std::string backend_name;
+    auto backend = gim.metadata(op_nh).get<cv::gimpl::FusedIsland>().object->backend();
+    std::string backend_impl_name = typeid(backend.priv()).name();
+
+    std::regex rgx("G([a-zA-Z0-9]+)BackendImpl");
+    std::smatch matches;
+    if (std::regex_search(backend_impl_name, matches, rgx)) {
+        GAPI_Assert(2ul == matches.size());
+        backend_name = matches[1].str();
+    }
```

Comment:
What if it doesn't match? The code shouldn't break because of that.
There's no strict backend naming scheme specified right now.
3rd party backends may miss this requirement.

Reply:
Fixed now, thanks!